util/bloom/bloom.go

```diff
@@ -90,6 +92,10 @@ func (f *Filter) Test(x []byte) bool {
 	return true
 }
 
+func (f *Filter) NumEntries() int {
```

Comment:
It's the other way around: you need to know the number of entries on *incoming* bloom filters, so you never had the "Set" function called there. instead, you should deduce that number during the decoding from the bloom filter size.

Reply:
I pack the number all the way through, added a field to the exchange encoded bloom filter

Reply:
for the xor filter it might be ok, but not for the bloom filter, since it would break existing clients.

Reply:
I'm also abit worried if that's open an attack vector.